base-containers/base/bin/INGInious

```diff
@@ -251,6 +254,19 @@ class INGIniousMainRunner(object):
         f.write(json.dumps(data))
         f.close()
 
+    def _detect_and_start_run(self):
+        """ By default, we run the `run` script under IPython. This function checks
```

Comment:
I'd rather make this explicit to avoid confusion and to ensure retro-compatibility the right way. This patch may make support harder in the future as problems will leverage from the fact that some users will sometimes forget this rule, or will simply be distracted and won't look at that line particularly. Just think about a missing "!" in the shebang. In the previous behaviour, Linux would simply say that this is not executable and this will be fixed very quickly. If this is therefore executed by ipython, this will be harder to debug for a common user.